diff --git a/bin/stale_check.sh b/bin/stale_check.sh
index 209b1a3..177b532 100755
--- a/bin/stale_check.sh
+++ b/bin/stale_check.sh
@@ -10,29 +10,28 @@
 for name in $( systemctl --full --all --plain --no-legend list-units scam_hls* | sed 's/scam_hls@\(\S*\).service.*/\1/' )
 do
 
-    echo $name
 
-    size=$( du /tmp/$1/ | gawk '{ print $1 }' )
+    size=$( du /tmp/$name/ | gawk '{ print $1 }' )
 
-    modtime=$( stat -c %Y /tmp/$1 )
+    modtime=$( stat -c %Y /tmp/$name )
     nowtime=$( date +%s )
     deltatime=$(( $nowtime - $modtime ))
 
 
+
     if [ "$size" -gt 10000 ]
     then
-        echo "$1 folder is too big: $size"
-        systemctl restart scam_hls@$1.service
+        echo "$name folder is too big: $size"
+        systemctl restart scam_hls@$name.service
     fi
 
 
     if [ "$deltatime" -gt 60 ]
     then
-        echo "$1 folder is too old: $deltatime seconds"
-        systemctl restart scam_hls@$1.service
+        echo "$name folder is too old: $deltatime seconds"
+        systemctl restart scam_hls@$name.service
     fi
 
 
 done
 
-
diff --git a/systemd/scam_heartbeat.service b/systemd/scam_heartbeat.service
index e869ee0..13440c7 100644
--- a/systemd/scam_heartbeat.service
+++ b/systemd/scam_heartbeat.service
@@ -2,9 +2,8 @@
 Description=Error Checker for the Cameras
 
 [Service]
-Type=simple
+Type=oneshot
 ExecStart=/opt/DogCamera/bin/stale_check.sh
-Restart=always
 
 [Install]
 WantedBy=multi-user.target

